src/pages/Wallet/RenameWallet.tsx

```diff
@@ -0,0 +1,114 @@
+import { useTranslation } from 'react-i18next';
+import {
+  Button,
+  CircularProgress,
+  Stack,
+  Typography,
+} from '@avalabs/core-k2-components';
+
+import { useApproveAction } from '@src/hooks/useApproveAction';
+import { ActionStatus } from '@src/background/services/actions/models';
+import { useGetRequestId } from '@src/hooks/useGetRequestId';
+import {
+  ApprovalSection,
+  ApprovalSectionBody,
+  ApprovalSectionHeader,
+} from '@src/components/common/approval/ApprovalSection';
+import { TxDetailsRow } from '@src/components/common/approval/TxDetailsRow';
+import {
+  AccountDetails,
+  WebsiteDetails,
+} from '../SignTransaction/components/ApprovalTxDetails';
+import { DomainMetadata } from '@src/background/models';
+
+export function RenameWallet() {
+  const { t } = useTranslation();
+  const requestId = useGetRequestId();
+
+  const { action, updateAction, cancelHandler } =
+    useApproveAction<{ walletId: string; newName: string }>(requestId);
+
+  console.log('action: ', action);
+
+  if (!action) {
+    return (
+      <Stack
+        sx={{
+          width: 1,
+          height: 1,
+          justifyContent: 'center',
+          alignItems: 'center',
+        }}
+      >
+        <CircularProgress size={60} />
+      </Stack>
+    );
+  }
+
+  console.log('action.displayData: ', action.displayData);
+
+  const site: DomainMetadata = action.site ?? {
+    domain: '#',
+    name: t('Unknown website'),
+  };
+
+  return (
+    <Stack sx={{ width: 1, px: 2 }}>
+      <Stack sx={{ flexGrow: 1, width: 1, gap: 3 }}>
+        <Typography variant="h4" sx={{ mt: 1.5, mb: 3.5 }}>
+          {t('Rename Wallet?')}
+        </Typography>
+        <ApprovalSection>
+          <ApprovalSectionHeader label={t('Action Details')} />
+          <ApprovalSectionBody sx={{ py: 1, px: 2, gap: 1 }}>
+            <WebsiteDetails site={site} />
+          </ApprovalSectionBody>
+        </ApprovalSection>
+        <ApprovalSection>
+          <ApprovalSectionHeader label={t('Account Details')} />
+          <ApprovalSectionBody sx={{ py: 1, px: 2, gap: 1 }}>
+            <AccountDetails
+              address={action.displayData.walletId}
+              label={t('Wallet ID to rename')}
+            />
```

Comment:
Instead of the `walletId` (which provides no information to the user), I think we should display the old wallet name.

Also, `<AccountDetails />` is not the proper component to use here - I think the simple `<TxDetailsRow />` will do better here.